### PR TITLE
added 3 endpoints to artist ts

### PR DESCRIPTION
**Pull request by K1llshot007**

### CODE REVIEW

File: src/endpoints/Artist.ts [hunk 1]
```diff
@@ -1,20 +1,58 @@
...
       queryParams: {},
     });
   }
+
+  /**
+   * Get Spotify catalog information about an artist's top tracks by country
+   * https://developer.spotify.com/documentation/web-api/reference/get-an-artists-top-tracks
+   */
+  getArtistTopTracks(id: string) {//also needs an optional param of market here
+      return this.get<TracksDetail>(`/artist/${id}/top-tracks`);//unsure how to join market with this ??
```
Comment (Abdullah-Malik) [inline src/endpoints/Artist.ts:46]: You are supposed to send the market param in the URL just like the following example

https://api.spotify.com/v1/artists/0TnOYISbd1XYRBk9myaseg/top-tracks?market=ES

File: src/endpoints/Artist.ts [hunk 2]
```diff
@@ -1,20 +1,58 @@
...
+   */
+  getArtists(ids: string[]) {
+    return this.get<ArtistsProfile>(`/artists?ids=${joinIdsArrayToString(ids)}`);
+  }
+
+  /**
+   * Get Spotify catalog information about an artist's albums
+   * https://developer.spotify.com/documentation/web-api/reference/get-an-artists-albums
+   */
   getArtistAlbums(artistId: string) {
```
Comment (Abdullah-Malik) [inline src/endpoints/Artist.ts:33]: Remove this endpoint and write it from scratch just like you have done for other endpoints

**Review by Abdullah-Malik — COMMENTED** [review]
Can you please write tests and documentation for artist before we merge it

**Comment by K1llshot007** [comment]
the required changes are complete, getArtistsAlbums endpoint is updated and new types added as required

File: tests/endpoints/Artist.test.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  describe('getArtistAlbums', () => {
+    it('should call get method with correct params and return expected result', async () => {
+      const mockArtistId = '1234';
+      const mockOptionalParams = { limit: 10, offset: 0 };
+      const mockResponse: ArtistAlbumResult = {
+        href: 'https://api.spotify.com/v1/artists/1234/albums',
+        items: [],
+        limit: 10,
+        next: 'https://api.spotify.com/v1/me/shows?offset=1&limit=1',
```
Comment (K1llshot007) [inline tests/endpoints/Artist.test.ts:69]: having some issues here, the next and previous can be nullable, but we cant assign null to a string

File: doc/endpoints/artist.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Return: `TracksDetail`
+
+The `TracksDetail` object contains:
+- `tracks`: An array of `Track` objects (see `getTrack` return type for `Track` object structure)
+
+#### Example
+
+```typescript
+const artist = await client.track.getTrack('artistId');
```
Comment (K1llshot007) [inline doc/endpoints/artist.md:132]: have to change examples based on the changes made in request maker and the get function

File: src/endpoints/Artist.ts [hunk 5]
```diff
@@ -1,19 +1,53 @@
...
+      return this.get<TracksDetail>(`/artist/${id}/top-tracks`, optionalParams);
+  }
+
+  /**
+   * Get Spotify catalog information about artists similar to a given artist.
+   * https://developer.spotify.com/documentation/web-api/reference/get-an-artists-related-artists
+   */
+  getRelatedArtists(id: string) {
+    return this.get<ArtistsProfile>(`/artists/${id}/related-artists`)
   }
```
Comment (K1llshot007) [inline src/endpoints/Artist.ts:51]: these will be changed according to the new get function

**Review by Abdullah-Malik — APPROVED** [review]

**Review by Abdullah-Malik — APPROVED** [review]

**Pull request merged**